internal/libs/mathslib/calc.go

```diff
@@ -29,3 +30,42 @@ func CalculateUsdPerUnitGas(sourceGasPrice *big.Int, usdPerFeeCoin *big.Int) *bi
 	tmp := new(big.Int).Mul(sourceGasPrice, usdPerFeeCoin)
 	return tmp.Div(tmp, big.NewInt(1e18))
 }
+
+// DeviatesOnCurve calculates a deviation threshold on the fly using xNew. For now it's only used for gas price
+// deviation calculation. It's important to make sure the order of xNew and xOld is correct when passed into this
+// function to get an accurate deviation threshold.
+func DeviatesOnCurve(xNew, xOld, noDeviationLowerBound *big.Int) bool {
+	// If xNew < noDeviationLowerBound, Deviates should never be true
+	if xNew.Cmp(noDeviationLowerBound) < 0 {
+		return false
+	}
+
+	xNewFloat := new(big.Float).SetInt(xNew)
+	xNewFloat64, _ := xNewFloat.Float64()
```

Comment:
This code is kind of odd, why not just use `BigFloat` everywhere and have peace of mind when it comes to these large gas price numbers and loss of precision / accuracy that compounds when doing floating point arithmetic?